plugins/reference-assets/core/src/assets/choice/transform.ts

```diff
@@ -0,0 +1,33 @@
+import type { TransformFunction } from '@player-ui/player';
+import type { ChoiceAsset, TransformedChoice } from './types';
+import { choicesEntryTransform } from './choices-entry'
+
+/**
+ * Choice asset transform
+ */
+export const choiceTransform: TransformFunction<ChoiceAsset, TransformedChoice> = (
+  asset,
+  options
+) => {
+  return {
+    ...asset,
+    set(val) {
```

Comment:
we don't need a `set` function for the whole choice itself, if the binding was updating from another expression, this would not get called, and the selection state would not update.


each individual `ChoiceEntry` should have a `select` function that sets its value into the data model for the binding

```typescript
const currentValue = options.data.model.get(binding)
const transformedChoices = asset.choices.map(choiceEntry => {
  return {
    ...choiceEntry,
    selected: choiceEntry.value === currentValue,
    select: () => {
        options.data.model.set([asset.binding, choiceEntry.value])
    }
  }
})

return {
  ...asset,
  choices: transformedChoices,
  // other properties like validation
}
```

Reply:
Just curious how this would work since I would still need to set the previously selected value as not selected, or is there a hook that updates all the selected values to choiceEntry.value === currentValue when the binding is set to a new value?

Reply:
you are never setting the selected value as not selected in this scenario, it would always use the data model as the source of truth, setting the value of the binding to a new value will rerender and the transform will run again and `selected` will be populated based on the new value in the data model